api/serializers.py

```diff
@@ -21,12 +23,21 @@ class Meta:
             'payload_request', 
             'related_ioc',
         )
-    
+
+class EnrichmentSerializer(IOCSerializer):
+    query = serializers.CharField(max_length=255, write_only=True)
+    found = serializers.BooleanField(read_only=True, default=False)
+    ico = IOCSerializer(read_only=True, default=None)
+
     def validate(self, data):
         """
         Check the given observable against regex expression
         """
         observable = data['observable_name']
         if not re.match(REGEX_IP, observable) or not re.match(REGEX_DOMAIN, observable):
-            raise serializers.ValidationError("Observable must be an IP or a domain")
-        return data
\ No newline at end of file
+            raise serializers.ValidationError("Observable is not a valid IP or domain")
+        try:
+            required_object = IOC.objects.get(name=observable)
+        except IOC.DoesNotExist:
+            raise serializers.ValidationError("Observable does not exist in the database")
```

Comment:
we do not want to raise exception, we want to mark `found` as False. so replace w/ `data["found"] = False`.